src/main/java/org/zalando/planb/provider/RealmConfig.java

```diff
@@ -74,10 +74,21 @@ void setup() {
                 .findFirst();
     }
 
+    static Optional<String> findRealmNameInRealm(@NotNull final Set<String> realmNames, @NotNull final String realm) {
+        return realmNames.stream()
+                .filter(r -> realm.contains(stripLeadingSlash(r)))
```

Comment:
I think "contains" is too dangerous as it would also match empty strings or single characters (or any substring)